drivers/lora/sx127x.c

```diff
@@ -331,94 +400,100 @@ static int sx1276_transceive(uint8_t reg, bool write, void *data, size_t length)
 			.count = ARRAY_SIZE(buf)
 		};
 
-		return spi_transceive(dev_data.spi, &dev_data.spi_cfg,
-				&tx, &rx);
+		return spi_transceive(dev_data.spi, &dev_data.spi_cfg, &tx, &rx);
 	}
 
 	return spi_write(dev_data.spi, &dev_data.spi_cfg, &tx);
 }
 
-int sx1276_read(uint8_t reg_addr, uint8_t *data, uint8_t len)
+int sx127x_read(uint8_t reg_addr, uint8_t *data, uint8_t len)
 {
-	return sx1276_transceive(reg_addr, false, data, len);
+	return sx127x_transceive(reg_addr, false, data, len);
 }
 
-int sx1276_write(uint8_t reg_addr, uint8_t *data, uint8_t len)
+int sx127x_write(uint8_t reg_addr, uint8_t *data, uint8_t len)
 {
-	return sx1276_transceive(reg_addr | BIT(7), true, data, len);
+	return sx127x_transceive(reg_addr | BIT(7), true, data, len);
 }
 
-void SX1276WriteBuffer(uint16_t addr, uint8_t *buffer, uint8_t size)
+void SX127xWriteBuffer(uint16_t addr, uint8_t *buffer, uint8_t size)
 {
 	int ret;
 
-	ret = sx1276_write(addr, buffer, size);
+	ret = sx127x_write(addr, buffer, size);
 	if (ret < 0) {
 		LOG_ERR("Unable to write address: 0x%x", addr);
 	}
 }
 
-void SX1276ReadBuffer(uint16_t addr, uint8_t *buffer, uint8_t size)
+void SX127xReadBuffer(uint16_t addr, uint8_t *buffer, uint8_t size)
 {
 	int ret;
 
-	ret = sx1276_read(addr, buffer, size);
+	ret = sx127x_read(addr, buffer, size);
 	if (ret < 0) {
 		LOG_ERR("Unable to read address: 0x%x", addr);
 	}
 }
 
-void SX1276SetRfTxPower(int8_t power)
+void SX127xSetRfTxPower(int8_t power)
 {
 	int ret;
 	uint8_t pa_config = 0;
 	uint8_t pa_dac = 0;
 
-	ret = sx1276_read(SX1276_REG_PA_DAC, &pa_dac, 1);
+	ret = sx127x_read(REG_PADAC, &pa_dac, 1);
 	if (ret < 0) {
 		LOG_ERR("Unable to read PA dac");
 		return;
 	}
 
 	pa_dac &= RF_PADAC_20DBM_MASK;
 
-	if (SX1276_PA_OUTPUT(power) == SX1276_PA_BOOST) {
+	if (SX127X_PA_OUTPUT(power) == SX127X_PA_BOOST) {
 		power = clamp_int8(power, 2, 20);
 
 		pa_config |= RF_PACONFIG_PASELECT_PABOOST;
 		if (power > 17) {
 			pa_dac |= RF_PADAC_20DBM_ON;
-			pa_config |= (power - 5) & 0x0F;
+			pa_config |= (power - 5) & RF_PACONFIG_OUTPUTPOWER_MASK;
 		} else {
 			pa_dac |= RF_PADAC_20DBM_OFF;
-			pa_config |= (power - 2) & 0x0F;
+			pa_config |= (power - 2) & RF_PACONFIG_OUTPUTPOWER_MASK;
 		}
 	} else {
+#ifdef RF_PACONFIG_MAX_POWER_MASK
 		power = clamp_int8(power, -4, 15);
 
 		pa_dac |= RF_PADAC_20DBM_OFF;
 		if (power > 0) {
 			/* Set the power range to 0 -- 10.8+0.6*7 dBm. */
-			pa_config |= 7 << SX1276_PA_CONFIG_MAX_POWER_SHIFT;
-			pa_config |= (power & 0x0F);
+			pa_config |= ~RF_PACONFIG_MAX_POWER_MASK;
+			pa_config |= (power & RF_PACONFIG_OUTPUTPOWER_MASK);
```

Comment:
`RF_PACONFIG_OUTPUTPOWER_MASK` is `0xF0`, but we used `0x0F` so far. Also `pa_config |= ~RF_PACONFIG_MAX_POWER_MASK;` line seems suspicious. Those mask macros are already inverted in loramac-node, which looks odd to me. The usual pattern is to do `reg &= ~mask;`, `reg |= bits;` and `reg |= (val & mask);`. What about creating helper macros like:
```
#define SX127X_PACONFIG_MAX_POWER_MASK (~RF_PACONFIG_MAX_POWER_MASK)
#define SX127X_PACONFIG_OUTPUTPOWER_MASK (~RF_PACONFIG_OUTPUTPOWER_MASK)
```
and use them with the patterns above?

Reply:
A better alternative from Zephyr point of view would be to post changes (inversion) of `RF_PACONFIG_MAX_POWER_MASK` and `RF_PACONFIG_OUTPUTPOWER_MASK` to mainline loramac-node. But that could potentially harm other projects. Maybe it is good to try and see what is the feedback there.

Reply:
Nice catch, thanks! These masks are weird and confusing, but I'm afraid they're public API for a long time. I'll create the helpers for now.

Reply:
I've wrapped these RF_ macros to the helper ones with more usual semantics. @mniestroj, could you please look again?